### DISCUSSION TITLE
OpenAPI annotation of BaseRoute?

### DISCUSSION THREAD

**Discussion by The-Funk** [quarkusio/quarkus#25377] [Q&A]
How do I specify to the smallrye openapi extension that I want to name a section in the spec after the BaseRoute and not just have it named "default"?

If I use JAX-RS and annotate my class with Path the generated spec includes my class name as the section name, but if I use Reactive Routes and annotate my class with BaseRoute, all of my routes under that base route are thrown in the "default" section.

[a comment by an automated account is not shown]

**Reply by phillip-kruger** [reply]
Hi. What version of Quarkus are you using ? The name should be defaulting to the classname, but you can also name it using the `@Tag` annotation, example `@Tag(name = "BaseRoute", description = "some description here")`

**Reply by The-Funk** [reply]
Latest, 2.8

The class was originally using JAX-RS but I both renamed it and rewrote it with the reactive routes annotations, because that style just jives with me more, and the name felt off. After removing @Path and replacing it with @BaseRoute and renaming the class and file in my IDE, the class name was no longer used as the Tag name. Applying @Tag with a name value worked just fine.

**Reply by phillip-kruger** [reply]
Ok, that seems like a bug then. It should be constant over JAX-RS, Spring Web and Reactive Routes. Also look at RestEasy Reactive, it think that is the way to go if you want reactive endpoints.

**Reply by The-Funk** [reply]
> Ok, that seems like a bug then. It should be constant over JAX-RS, Spring Web and Reactive Routes. Also look at RestEasy Reactive, it think that is the way to go if you want reactive endpoints.

I had been using RestEasy Reactive, that's where the original JAX-RS style annotations came from, but I generally prefer Sinatra-esque syntax over JAX-like annotations, so I added the Reactive Routes extension. Sorry, I should have been more clear in my original post.
